File: test_archives/py_tests/batch1/test_function3.py
from functions.function3 import function3
import pytest
def test_function3_1():
    assert function3('a', 1, {'x': 2, 'y': 3}, True) == 'a15True'

def test_function3_2():
    assert function3('b', 2, {'x': 4, 'y': 5}, False) == 'b29False'

File: test_archives/cpp_tests/batch1/test_function3.cpp
#include "../catch.hpp"
#include "../functions/function3.cpp"

TEST_CASE( "function3_1 tests", "[function3]" ) {
    REQUIRE( function3('a', 1, {'x': 2, 'y': 3}, True) == "a15True" );
}

TEST_CASE( "function3_2 tests", "[function3]" ) {
    REQUIRE( function3('b', 2, {'x': 4, 'y': 5}, False) == "b29False" );
}